src/librbd/internal.cc

```diff
@@ -2132,13 +2205,32 @@ namespace librbd {
     // ignore return value, since we may be set to a non-existent
     // snapshot and the user is trying to fix that
     ictx_check(ictx);
+    if (ictx->image_watcher != NULL) {
+      ictx->image_watcher->flush_aio_operations();
```

Comment:
I think we'll need to do this everywhere we flush the rbd cache - maybe it would make sense as part of that method in addition to being used here